### ISSUE TITLE
Server-side code doesn't refresh when changed

### ISSUE THREAD

**Issue by viridia** [solidjs/solid-start#454]
When I make a change to a server-side module, the solid-start server restarts, but doesn't actually load the new code. I have to manually ctrl-c the server process and re-run `npm run dev` to see the new behavior.

This only appear to affect some modules and not others. Changing code inside of a `createServerData$()` block does get refreshed as expected. Changing the code of a graphql resolver, on the other hand, does not.

Certain files, when touched, will cause the entire server to refresh. For example:

* I add a 'console.log('aaa') to a graphql resolver and hit save.
* Server restarts, but I do not see 'aaa' printed on the terminal.
* Now, I go to another file that has a `createServerData$` in it, and hit save.
* Server restarts again, only now I see the 'aaa' printed.

In other words, touching certain files causes the whole server to reload.

**Comment by ryansolid** [comment]
This is going to be hard to track down. Going to need a reproduction.

**Comment by viridia** [comment]
In terms of providing a reproduction: what's acceptable? I realize that a stackblitz url would be ideal for you, but that's a lot of work for me, since the problem is unlikely to manifest in small projects and I'd have to port my existing project to stackblitz. OTOH, I could point you to a github repo of my existing solid-start project, but you'd have to run docker to get it up and running, etc. That's a lot to ask I imagine. What's an acceptable middle ground?

It's been a while since I even looked at the project which was exhibiting this problem. Of the three solid-start projects I have, this one is the most idiomatic, meaning that it doesn't deviate very much from the standard solid-start pattern. The biggest change from vanilla solid-start is the addition of graphql and prisma.

**Comment by illispi** [comment]
I don't have a stackblitz either but with my repo it should repro pretty easily.

1. download this https://github.com/illispi/stories/tree/0571d54659eab98c5757ab70254620eb2a5027e7
2. go to that folder and pnpm install
3. cd ./apps/solid-stories
4. pnpm dev
5. Open the localhost:3000
6. open ./src/server/server.ts
7. Edit for instance, const getSessionFromCookie = async (request: Request) => {} by adding console.log("Should fail to reload") inside it.
8. Refresh the page in browser and "Should fail to reload" doesnt appear in cmd line, unless you ctrl+C and run pnpm dev again.

**Comment by spiffytech** [comment]
I'm hitting an issue and I'm not sure if it's the same issue, or a different one:

When a server-side module changes, that file gets reloaded, but nothing that file imports gets reloaded.

This leaves code holding outdated values imported from previous versions of the code. Logic that expects a clean environment doesn't have one.

I ran into this when the Prometheus metrics `prom-client` library emitted an error regarding duplicate metrics, but it seems to affect any module-scoped logic or data.

## Reproduction

`entry-server.tsx`
```tsx
import {
  createHandler,
  renderAsync,
  StartServer,
} from "solid-start/entry-server";

import { print } from "./lib/lib-a.js";

setInterval(print, 1_000);

export default createHandler(
  renderAsync((event) => <StartServer event={event} />)
);
```

`lib/lib-a.ts`
```ts
import { globalFoo } from "./lib-b";

console.log("Running lib-a", { globalFoo });

let foo = new Date();
export const print = () => console.log(foo, globalFoo);
```

`lib/lib-b.ts`
```ts
console.log("Running lib-b");
export const globalFoo = new Date();
```

If you edit `lib/lib-a.ts`, you'll see that the `Date` imported from `lib/lib-b.ts` never changes - lib-b's export is never recreated.

This stale value issue becomes worse with the use of timeouts/intervals. If you make changes to the `print()` function exported by `lib/lib-a.ts`, `entry-server.tsx` gets reloaded and creates a new interval, resulting in both the old and new `print()` function bodies getting executed.

(In addition to the stale value issue, I'm not sure how I'm supposed to handle cleanup of stuff like intervals if only part of the server is relaunched)

**Comment by viridia** [comment]
That matches what I am seeing, and narrows the problem much better.

**Comment by spiffytech** [comment]
@ryansolid With the above reproduction, can we please remove the `needs reproduction` issue tag?

I assume the Astro integration will upend this kind of bug. Should we expect to see this issue put on hold until after that comes out?

**Comment by ryansolid** [comment]
Hmm... this issue may or may not exist after rebasing. The handling of server functions is completely custom and it must not hook into all the things it needs to. Thanks for the reproduction.

**Comment by ryansolid** [comment]
In setting up for SolidStarts next Beta Phase built on Nitro and Vinxi we are closing all PRs/Issues that will not be merged due to the system changing. If you feel your issue was closed in mistake. Feel free to re-open it after updating/testing against 0.4.x release. Thank you for your patience.  See https://github.com/solidjs/solid-start/pull/1139 for more details.
